src/test/java/dev/zarr/zarrjava/ZarrTest.java

```diff
@@ -278,35 +276,31 @@ public void testCheckInvalidChunkBounds(int[] chunkSize) throws Exception {
         assertThrows(ZarrException.class, builder::build);
     }
 
+    static Stream<int[]> invalidShardSizes() {
+        return Stream.of(
+            new int[]{4}, //wrong dims
+            new int[]{4, 4, 4}, //wrong dims
+            new int[]{1, 1}, //smaller than inner chunk shape
+            new int[]{5, 5} //no exact multiple of inner chunk shape
```

Comment:
:+1: Only additional suggestion would be to also include `new int[] {2, 1}` and `new int[]{2, 5}` to the invalid shard sizes scenarios.
This should test an exception raised when the divisibility rule is not met for the second dimension.